packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -722,188 +537,157 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
+     * @param _managerSellAmount quantity the manager wants to sell
+     * @param _clearingPrice the price at which the manager is buying
+     * @param _orders an array of signed order to swap tokens
      */
-    function _startAuction(uint256 _auctionTriggerTime)
-        internal
-        returns (
-            bool,
-            uint256,
-            uint256,
-            uint256
-        )
-    {
-        (uint256 strategyDebt, uint256 ethDelta) = _syncStrategyState();
-        uint256 currentWSqueethPrice = IOracle(oracle).getTwap(
-            ethWSqueethPool,
-            wPowerPerp,
-            weth,
-            hedgingTwapPeriod,
-            true
-        );
-        uint256 feeAdjustment = _calcFeeAdjustment();
-        (bool isSellingAuction, ) = _checkAuctionType(strategyDebt, ethDelta, currentWSqueethPrice, feeAdjustment);
-        uint256 auctionWSqueethEthPrice = _getAuctionPrice(_auctionTriggerTime, currentWSqueethPrice, isSellingAuction);
-        (bool isStillSellingAuction, uint256 wSqueethToAuction) = _checkAuctionType(
-            strategyDebt,
-            ethDelta,
-            auctionWSqueethEthPrice,
-            feeAdjustment
-        );
+    function hedgeOTC(
+        uint256 _managerSellAmount,
+        uint256 _clearingPrice,
+        Order[] memory _orders
+    ) external onlyOwner {
+        require(_clearingPrice > 0, "Manager Price should be greater than 0");
+        require(_isTimeHedge() || _isPriceHedge(), "Time or Price is not within range");
+        _checkOTCPrice(_clearingPrice, _orders[0].managerToken);
 
-        require(isSellingAuction == isStillSellingAuction, "auction direction changed");
+        timeAtLastHedge = block.timestamp;
 
-        uint256 ethProceeds = wSqueethToAuction.wmul(auctionWSqueethEthPrice);
+        uint256 remainingAmount = _managerSellAmount;
+        uint256 prevPrice = 0;
+        uint256 currentPrice = 0;
+        bytes memory tradePair;
+        bytes memory prevTradePair;
+        for (uint256 i = 0; i < _orders.length; i++) {
+            tradePair = abi.encode(_orders[i].traderToken, _orders[i].managerToken);
+            currentPrice = _orders[i].managerAmount.mul(1e18).div(_orders[i].traderAmount);
+            require(currentPrice >= prevPrice, "Orders are not arranged properly");
+            if (i > 0) {
+                require(
+                    keccak256(tradePair) == keccak256(prevTradePair),
+                    "All orders must have the same buy/sell token."
+                );
+            }
+            prevPrice = currentPrice;
+            prevTradePair = tradePair;
+
+            if (remainingAmount > _orders[i].managerAmount) {
+                _execOrder(remainingAmount, _clearingPrice, currentPrice, _orders[i]);
+                remainingAmount = remainingAmount.sub(_orders[i].managerAmount);
+            } else {
+                _execOrder(remainingAmount, _clearingPrice, currentPrice, _orders[i]);
+                break;
```

Comment:
Might want a view function that validates an order and tells the MM what will happen to it?